telemetry/specs/INT.mdk

```diff
@@ -222,68 +222,27 @@ There are two types of INT Headers, *hop-by-hop* and *destination*.  A given INT
-system would be able to process INT instructions with no increase in latency or
-reduction in forwarding performance, but in some cases it may be required to
-process INT packets outside the fast path.  This slow-path processing could be a
-CPU based control plane, some sideband or alternate hardware assisted forwarding
-path, or an arbitrary INT resource.  Note that in the case where the INT
-processing is done outside the fast path, the device MUST still forward the
-original packet through the fast path (i.e. without processing the INT
-instructions).  The implementation of this is not specified, though it implies
-the ability to make a copy of the INT packet for slow-path processing or a
-similar design.  Following the processing of the INT packet in the normal
-fast-path, the forwarding plane should generate a **_trigger_** toward the slow
-path (e.g., either a copy of the original INT packet or a digest of it). Upon
-receiving the trigger, the slow path should process the INT Header
-appropriately, generate a new packet -- called a “follow-up packet” --
-containing the execution results of the INT instructions. The follow-up packet
-is forwarded separately.
-
-If devices in the network do perform slow-path INT processing, it is possible
-that a single INT packet could spawn multiple follow-up packets – and in turn
-each of these could spawn more INT processing downstream.  Care must be taken to
-prevent excessive replication.  To prevent the cascading generation of follow-up
-telemetry packets,  all follow-up packets are marked with a special
-“exemption” flag.  The presence of this flag instructs downstream devices to
-provide specific processing.  For more specific information, see the option type
-for INT Source-to-sink communication messages (Destination type in Section
-[#sec-int-header-types]).
 
-The INT Header of a follow-up packet must contain all the existing INT metadata
-in the original packet that was added by the upstream devices, as well as its
-own local INT metadata. Follow-up packets must contain enough information (from
-outer header, inner header, or both of the original packet) so that the INT sink
-can correlate the follow-up packets with the original INT packet. Because INT is
-not tied to a particular encapsulation protocol, this spec does not dictate the
-exact format of a follow-up packet other than its INT portion.
+The data format of Destination type INT Headers is not defined in this version,
+but is expected to be defined in future versions of this specification.
 
-To prevent potential packet ordering issues, it is recommended that an INT
-device **_NOT_** forward the INT packets themselves via the slow path while
-processing INT Headers.
+More INT Header types may be defined in future versions, for example to deliver
+piggybacked hop-by-hop metadata, from the original INT Sink back to the original
+INT Source for purposes of closed loop control.
```

Comment:
How is this different from destination type INT header?

Reply:
The contents are different. For piggybacked metadata, the format is exactly the same as the INT Hop-by-Hop Metadata Header, it just needs an indication (proposing different INT type) that this is to be passed through and not added to each hop.
For destination type INT header with contents such as sequence number or original IP TTL, these are contents that differ from the contents of the INT Hop-by-Hop Metadata Header.

Reply:
If I understand and recall correctly, exact definition of src-to-sink INT message was considered outside the scope of the spec. All this spec needs to define is two types of INT headers: hop-by-hop type and destination type. The format of destination type INT message only needs to be understood by the INT source and sink and different implementations could use their own formats, in a way this is "opaque" data. The destination type INT message could carry piggybacked metadata, or it could carry sequence numbers, or original TTL or something totally different.

Perhaps, just define two types, type 1 (hop-by-hop) and type 2 (destination type) and leave out this one sentence about more INT header types?